### PR TITLE
MB-6752 Admin UI: Add userId field to Admin & Office Users pages

### PR DESCRIPTION
**Pull request by sandy-wright**
## Description

This PR adds the `User Id` field to the Admin & Office Users list, detail, and edit pages. It also does some extra clean-up work around this part of the app:

* Adds full cypress tests for all pages for Admin & Office Users.
* Moves the `AdminUsers` and `OfficeUsers` folders from the deprecated `src/scenes/SystemAdmin` directory to the new standard `src/pages/Admin` directory.
* Adds an `.eslintrc` file to the `src/pages/Admin` directory that ignores the `react/jsx-no-props-spreading` rule for all files in the directory. We have this rule ignored in literally every page for the Admin UI, and we really should just ignore it for the whole Admin folder.
* Cleans up some miscellaneous bugs in the admin interface, such as removing the checkboxes that do nothing in the moves list, fixing a typo on the `middleInitials` field for office users, and updating some imports.

## Setup

Run `make e2e_test` to run the cypress tests (`cypress/integration/admin/adminUsers.js` and `officeUsers.js` are the new ones).

To test the interface, run:

```sh
make server_run
make admin_client_run
```

Select "Local Sign-in." Verify that you can see the "User Id" field populated on these pages:
* Office Users list page (the home screen)
* Office Users detail page
* Office Users edit page (as a disabled input)
* Admin Users list page
* Admin Users detail page
* Admin Users edit page (as a disabled input)

Try grabbing the user ID value from a record or two and search for it in the Users list page. You should find the record you want! 

## Code Review Verification Steps

* [ ] Request review from a member of a different team.
* [x] Have the Jira acceptance criteria been met for this change?

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-6752) for this change.

## Screenshots

Adding the "User Id" field to the Admin & Office Users list, detail, and edit pages:
Screen Shot 2021-03-03 at 11 28 17 AM
Screen Shot 2021-03-03 at 11 28 27 AM
Screen Shot 2021-03-03 at 11 28 56 AM
Screen Shot 2021-03-03 at 11 29 13 AM

Removing the superfluous checkboxes from the Moves list page:
Screen Shot 2021-03-03 at 11 29 35 AM

### CODE REVIEW

File: src/pages/Admin/.eslintrc [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "rules": {
+    // React-Admin relies heavily on props spreading to handle all the default machinery under the hood.
+    // We therefore can't reasonably enable this rule for these pages.
+    "react/jsx-props-no-spreading": "off"
```
Comment (eamahanna) [inline src/pages/Admin/.eslintrc:5]: Did you confirm that this is okay to do with ATO requirements?

File: cypress/integration/admin/officeUsers.js [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+describe('Office Users List Page', function () {
+  before(() => {
+    cy.prepareAdminApp();
+  });
+
+  it('successfully navigates to Office Users page', function () {
+    cy.signInAsNewAdminUser();
+    // we should be at the office users list page by default,
+    // but let's go somewhere else and then come back to make sure the side nav link works
+    cy.get('a[href*="system/moves"]').click();
```
Comment (eamahanna) [inline cypress/integration/admin/officeUsers.js:12]: nice

File: src/pages/Admin/Moves/MoveList.jsx [hunk 3]
```diff
@@ -1,20 +1,26 @@
...
+import { Datagrid, DateField, Filter, List, TextField, TextInput } from 'react-admin';
 
 import AdminPagination from 'scenes/SystemAdmin/shared/AdminPagination';
-import styles from 'scenes/SystemAdmin/Home.module.scss';
 
 const defaultSort = { field: 'locator', order: 'ASC' };
 
 const MoveFilter = (props) => (
-  <Filter {...props} className={styles['system-admin-filters']}>
+  <Filter {...props}>
```
Comment (eamahanna) [inline src/pages/Admin/Moves/MoveList.jsx:10]: Why was the class name removed?

File: src/pages/Admin/OfficeUsers/OfficeUserCreate.jsx [hunk 4]
```diff
@@ -1,13 +1,14 @@
...
 import { RolesCheckboxInput } from 'scenes/SystemAdmin/shared/RolesCheckboxes';
 import { phoneValidators } from 'scenes/SystemAdmin/shared/form_validators';
-import { Create, SimpleForm, TextInput, ReferenceInput, AutocompleteInput, required } from 'react-admin';
 
 const OfficeUserCreate = (props) => (
   <Create {...props}>
     <SimpleForm>
       <TextInput source="firstName" validate={required()} />
-      <TextInput source="middle_initial" />
+      <TextInput source="middleInitials" />
```
Comment (eamahanna) [inline src/pages/Admin/OfficeUsers/OfficeUserCreate.jsx:11]: Nice catch

**Review by eamahanna — APPROVED** [review]
LGTM :)

File: src/pages/Admin/.eslintrc (hunk 1 above)
Comment (sandy-wright) [reply src/pages/Admin/.eslintrc:5]: I just got confirmation that it's fine, woo! https://ustcdp3.slack.com/archives/CTQQJD3G8/p1614963697014000?thread_ts=1614793450.005800&cid=CTQQJD3G8

File: src/pages/Admin/Moves/MoveList.jsx (hunk 3 above)
Comment (sandy-wright) [reply src/pages/Admin/Moves/MoveList.jsx:10]: There's already a filter class applied by default, `filter-field`. This is a custom style that fiddles with the padding and line height, and it causes issues with label placement and the font size is a lot less readable. I couldn't see any reason to use it, and we don't on many pages, so I just removed it for consistency's sake.

**Pull request merged**